deployments/workloads.go

```diff
@@ -229,15 +246,45 @@ func (w Workloads) createGiteaCredsSecret(c *kubernetes.Cluster, options kuberne
 }
 
 func (w Workloads) createWorkloadsServiceAccountWithSecretAccess(c *kubernetes.Cluster) error {
-	automountServiceAccountToken := false
+	automountServiceAccountToken := true
 	_, err := c.Kubectl.CoreV1().ServiceAccounts(WorkloadsDeploymentID).Create(
 		context.Background(),
 		&corev1.ServiceAccount{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: WorkloadsDeploymentID,
 			},
+			Secrets:                      []corev1.ObjectReference{{Name: "gitea-creds"}},
```

Comment:
Same here: why is this change needed now ? what changed and what does the change provide ?

Reply:
Same answer as before. Regarding if the change is needed and what it does provide it is stated on the commit/pr message. Basically, as we have been using the triggers account we are giving a lot of unnecessary permissions.

Reply:
Just for the sake of clarity, this is part of the convention used by Tekton to authenticate with git/docker: https://github.com/tektoncd/pipeline/blob/main/docs/auth.md#configuring-basic-auth-authentication-for-git